docs/COREDUMP_PACKAGE_METADATA.md

```diff
@@ -67,6 +67,14 @@ Value: a JSON string with the structure described below
 }
 ```
 
+When it comes to JSON numbers, this specification assumes that JSON parsers
+processing this information are capable of reproducing the full signed and
+unsigned 64bit integer range (i.e. -2⁶³…+2⁶⁴-1) as well as the full 64bit IEEE
```

Comment:
The full 64-bit range is definitely not supported everywhere:

```javascript
const json = '{"number": 9007199254740993}';
const obj = JSON.parse(json);
console.log(obj.number);
```
This comes back as: `9007199254740992`.

So you should restrict this to 52-bit integers with a separate sign.

And of course many implementations are not able to round-trip floating pointer numbers accurately through decoding and encoding (either changing the textual representation to something ugly, or not encoding the value accurately).

Reply:
I doubt we'll ever need all of that for this purpose, so IMHO we should specify the lowest common denominator among the current and popular oss implementations

Reply:
I htink we have the liberty here to make choice. If this was somethin we expect to be consumed by JS then yeah, I think we should make this more restrictive. But given this is mostly expected to be processed by system-level stuff, and 64bit integers are kinda a given for that these days I'd not limit things needlessly here. but dunno, i am fine with anything as long as we document it.

if we didn't document it then the only thing people reasonably can expect is the 52bit integer thing. But if we declare it I think we can be more free.

Reply:
(the json libs i checked are all system level stuff, C, python, glib, and so on)

Reply:
The node.js tooling may need to generate/process this data, too. Javascript isn't just for browsers anymore.

Just restrict it to 52 bits, please.